### PR TITLE
MAX_INDEX_STRING_LENGTH

### PR DESCRIPTION
**Pull request by fogelito**

### CODE REVIEW

File: tests/Database/Base.php [hunk 1]
```diff
@@ -433,11 +433,12 @@ public function testCreateCollectionValidator()
             "validator.test",
         ];
 
+
         $attributes = [
             new Document([
                 '$id' => ID::custom('attribute1'),
                 'type' => Database::VAR_STRING,
-                'size' => 2500, // longer than 768
+                'size' => Database::MAX_INDEX_STRING_LENGTH + 2000,
```
Comment (christyjacob4) [inline tests/Database/Base.php:441]:
```suggestion
                'size' => Database::MAX_INDEX_STRING_LENGTH + 2000, // longer than 768
```

**Review by christyjacob4 — APPROVED** [review]

File: src/Database/Database.php [hunk 2]
```diff
@@ -32,6 +32,8 @@ class Database
     const INDEX_SPATIAL = 'spatial';
     const INDEX_ARRAY = 'array';
 
+    const MAX_INDEX_STRING_LENGTH = 768; // mysql limit
```
Comment (TorstenDittmann) [inline src/Database/Database.php:35]: Wouldn't it make more sense to add a Method to each adapter for this?

Comment (abnegate) [reply src/Database/Database.php:35]: @fogelito Please follow up on this, adding a `getMaxIndexStringLength` function to each adapter

**Comment by abnegate** [comment]
Closing this, to be addressed in a follow up task

**Pull request closed without merging**